chaindexing-core/src/events.ts

```diff
@@ -1,14 +1,21 @@
 export interface Event {
   contractAddress: string;
   contractName: string;
+  chainId: number;
   abi: string;
-  logParams: Map<String, String>;
-  parameters: Map<String, String>;
-  topics: Map<String, String>;
+  logParams: Record<string, string>;
+  parameters: Record<string, string>;
+  topics: Record<string, string>;
   blockHash: string;
   blockNumber: number;
   transactionHash: string;
   transactionIndex: number;
   logIndex: number;
+  blockTimestamp: number;
   removed: boolean;
 }
+
+export interface SavedEvent extends Event {
```

Comment:
Let's keep the interfaces consistent: `UnsavedEvent` and `Event`, vs `SavedEvent` and `Event`. 

I strongly prefer `Unsaved...` because it is an edge-case concern and could be dangerous to be thought of as
the default `Event` model.